### ISSUE TITLE
Php Unit test case setup

### ISSUE THREAD

**Issue by DevikVekariya** [rtCamp/amp-admanager#60]
- Create a setup for Php Unit Tests along with the documentation.

Acceptance Criteria:

- Each class and its method should pass though the Php Unit test case
- If any of the methods are ignored then, it should have proper docs with a reason to ignore it.
- It should have detailed documentation for the local setup of Unit tests and how to run it locally.

Can take a reference from here - https://github.com/rtCamp/wp-google-login/pull/19

**Comment by dishitpala** [comment]
@DevikVekariya 
I need 24hrs to complete the task 
Which includes learning and implementation
There 3 different classes in with 17 unique functions 
The test will cover all the classes and the functions

Time estimation may vary because I will be starting learning as well as implementation completely from scratch

**Comment by DevikVekariya** [comment]
@dishitpala  I wanted to split this issue between multiple folks.

- Create a new branch `dev-unit-test` from master.
- Create a new branch from the dev-unit-test to write the test case
- Setup the unit tests setup from reference https://github.com/rtCamp/wp-google-login/pull/19
- **Raise the PR from your test-case branch to `dev-unit-test`**
- **Make sure that we'll have one PR from dev-unit-test to master for all together**

--------------------------------
- [ ] Setup the unit tests from https://github.com/rtCamp/wp-google-login/pull/19 => @dishitpala 
Please add and merge ☝️above setup in this repo , so multiple folks can work on same parallely
- [ ] classes/admin.php => @dishitpala  
- [ ] classes/amp-admanager.php => @tbd
- [ ] classes/shortcode.php => @tbd

Assign PRs to @deepaklalwani97 for the Review

**Comment by thrijith** [comment]
@deepaklalwani97 Please review and merge https://github.com/rtCamp/amp-admanager/pull/63 if it looks ok to you. Thanks

**Comment by thrijith** [comment]
The setup has been merged into https://github.com/rtCamp/amp-admanager/tree/dev-unit-test, tests for each classes can now be created from that branch as base.

**Comment by thrijith** [comment]
Added case for `classes/admin.php` in https://github.com/rtCamp/amp-admanager/blob/add/admin-class-tests/tests/classes/test-class-admin.php, working on `classes/shortcode.php` currently.

**Comment by thrijith** [comment]
Done with assigned classes for the plugin.
cc: @DevikVekariya @maitreyie-chavan

**Comment by vaishaliagola27** [comment]
Added test case for ` classes/amp-admanager.php`
Please review https://github.com/rtCamp/amp-admanager/pull/71.

**Comment by vaishaliagola27** [comment]
@DevikVekariya I've merged PR related to `classes/amp-admanager.php` tests.

**Comment by deepaklalwani97** [comment]
@juhi123 We have added unit tests for this repo in `dev-unit-test` and the branch needs to be QA tested before we can merge it. Please follow the readme and test the `dev-unit-test` branch.
